spark/src/main/java/org/apache/iceberg/spark/data/vectorized/ConstantColumnVector.java

```diff
@@ -109,6 +109,9 @@ public Decimal getDecimal(int rowId, int precision, int scale) {
 
   @Override
   public UTF8String getUTF8String(int rowId) {
+    if (constant instanceof String) {
```

Comment:
interesting! why is this special handling needed?

Reply:
Yeah it's weird, it seems spark vectorized API expect a string type to be `UTF8String`, so when we have a default value object of String type, we need to convert explicitly to `UTF8String`

Reply:
I see. Thanks for the explanation!